tsdb/tx.go

```diff
@@ -248,15 +254,27 @@ func (l *LocalMapper) Open() error {
 	l.txn = txn
 
 	// create a bolt cursor for each unique series id
-	l.cursors = make([]*bolt.Cursor, len(l.seriesKeys))
+	l.cursors = make([]*shardCursor, len(l.seriesKeys))
 
 	for i, key := range l.seriesKeys {
+		// Retrieve key bucket.
 		b := l.txn.Bucket([]byte(key))
-		if b == nil {
+
+		// Ignore if there is no bucket or points in the cache.
+		if b == nil && len(l.shard.cache[key]) == 0 {
 			continue
 		}
 
-		l.cursors[i] = b.Cursor()
+		// Retrieve a copy of the in-cache points for the key.
+		cache := make([][]byte, len(l.shard.cache[key]))
+		copy(cache, l.shard.cache[key])
```

Comment:
That's why I say that flushing should be supported on a per series basis. Some series are irregular so 20m could be a ton of data, but for series that are only sampled every minute, 20m is only 20 data points.

We need to be able to have good support for both high frequency and low frequency time series, so I don't think it makes sense to have it behave the same across the entire WAL.